app/jobs/create_backup_job.rb

```diff
@@ -0,0 +1,46 @@
+class CreateBackupJob < ActiveJob::Base
+  queue_as :default
+
+  def perform(backup)
+    unpack_backup_vars(backup)
+
+    return unless service.commands["backup"]
+
+    ensure_backup_directory
+
+    @backup.update(filename: backup_name)
+    run_backup
+    @backup.update(running: false)
+  end
+
+  private
+
+  attr_accessor :backup, :service, :app
+
+  def backup_name
+    @backup_name ||= "#{backup.app.clean_name}-#{service.name}-#{DateTime.now.to_s.parameterize}"
+  end
+
+  def run_backup
+    SshExecution.new(app.server).execute(command: backup_command)
+    SshExecution.new(app.server).scp(from: "/root/#{backup_name}",
+                                     to: Rails.root.join("backups", "#{app.clean_name}", backup_name))
+    SshExecution.new(app.server).execute(command: "rm #{backup_name}")
```

Comment:
I have no knowledge of the mechanics of `SshExecution` but I assume they execute subsequently? I mean they don't fire off a command to the server and then just continue with the next?